Compiler warning about babel-preset-es2015@6.24.1

happens during `npm install`

```
npm WARN deprecated babel-preset-es2015@6.24.1: We're super 😸  excited that you're trying to
use ES2015 syntax, but instead of continuing yearly presets 😭 , we recommend using 
babel-preset-env: npm install babel-preset-env. preset-env without options will compile ES2015+
down to ES5. And by targeting specific browsers, Babel can do less work and you can ship native 
ES2015+ to users 😎 ! Also, we are in the process of releasing v7, so give 
http://babeljs.io/blog/2017/09/12/planning-for-7.0 a read and test it! Thanks so much for using 
Babel 🙏 , please give us a follow @babeljs for updates, join slack.babeljs.io for discussion and help 
support at opencollective.com/babel
```

Thank you, @tiffon! I didn't have an issue with flow, re the issue mentioned. I had thought perhaps it had to do with the ordering of the plugins, but I'm not getting that error, with `transform-class-properties` ordered first or second. If it's still an issue, I'd need more information.

On the other hand, I am having a Travis failure, with error: `Error: Cannot find module 'process' (While processing preset: "/home/travis/build/jaegertracing/jaeger-client-node/node_modules/babel-preset-env/lib/index.js")`

--- a/.babelrc
+++ b/.babelrc
@@ -1,4 +1,4 @@
 {
-  "presets": ["es2015"],
-  "plugins": ["transform-flow-strip-types", "transform-class-properties"]
+  "presets": ["env"],
+  "plugins": ["transform-class-properties", "transform-flow-strip-types"]
 }

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,24 +3,7 @@ sudo: required
 language: node_js
 
 node_js:
-- 'node'
-- '8'
-- '6'
 - '4'
-- '0.12'
-
-matrix:
-  include:
-  - node_js: '0.10'
-    env: COVER=1
-    sudo: required
-    services:
-    - docker
-  - node_js: '0.10'
-    env: CROSSDOCK=1
-    sudo: required
-    services:
-      - docker
 
 cache:
   directories:
@@ -28,6 +11,21 @@ cache:
 
 services:
 - docker
+
+matrix:
+  include:
+  - env: TEST_NODE_VERSION=0.12
+  - env: TEST_NODE_VERSION=4
+  - env: TEST_NODE_VERSION=6
+  - env: TEST_NODE_VERSION=node
+  - env: COVER=1 TEST_NODE_VERSION=0.10
+    sudo: required
+    services:
+    - docker
+  - env: CROSSDOCK=1 TEST_NODE_VERSION=0.10
+    sudo: required
+    services:
+      - docker
 
 env:
   global:
@@ -44,9 +42,32 @@ before_install:
 - npm install babel-cli
 - if [ "$CROSSDOCK" = "1" ]; then make install_docker_ci ; fi
 
+before_script:
+# Install the Node version to test against
+- nvm install $TEST_NODE_VERSION
+- nvm use $TRAVIS_NODE_VERSION
+
 script:
-- if [ "$CROSSDOCK" != "1" ]; then make test ; fi
-- if [ "$COVER" = "1" ]; then npm run coveralls ; fi
+- if [ "$CROSSDOCK" != "1" ]; then
+    set -e;
+    make build-node;
+    nvm use $TEST_NODE_VERSION;
+    node --version;
+    rm -rf ./node_modules package-lock.json;
+    npm install;
+    make test-without-build;
+    nvm use $TRAVIS_NODE_VERSION;
+    fi
+- if [ "$COVER" = "1" ]; then
+    set -e;
+    make build-node;
+    nvm use $TEST_NODE_VERSION;
+    node --version;
+    rm -rf ./node_modules package-lock.json;
+    npm install;
+    npm run coveralls;
+    nvm use $TRAVIS_NODE_VERSION;
+    fi
 - if [ "$CROSSDOCK" = "1" ]; then make crossdock-fresh ; fi
 
 after_failure:

--- a/.travis.yml
+++ b/.travis.yml
@@ -64,7 +64,7 @@ after_failure:
 - if [ "$CROSSDOCK" = "1" ]; then timeout 5 make crossdock-logs ; fi
 
 after_success:
-- if [ "$CROSSDOCK" = "1"] && ["$TRAVIS_PULL_REQUEST" == false]; then
+- if [ "$CROSSDOCK" = "1" ]; then
     export REPO=jaegertracing/xdock-node;
     export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi);
     export TAG=`if [ "$BRANCH" == "master" ]; then echo "latest"; else echo $BRANCH; fi`;

--- a/.travis.yml
+++ b/.travis.yml
@@ -41,34 +41,24 @@ before_install:
 - docker version
 - npm install babel-cli
 - if [ "$CROSSDOCK" = "1" ]; then make install_docker_ci ; fi
-
-before_script:
 # Install the Node version to test against
 - nvm install $TEST_NODE_VERSION
 - nvm use $TRAVIS_NODE_VERSION
 
+before_script:
+- make build-node
+- nvm use $TEST_NODE_VERSION
+- node --version
+- rm -rf ./node_modules package-lock.json
+- npm install
+
 script:
-- if [ "$CROSSDOCK" != "1" ]; then
-    set -e;
-    make build-node;
-    nvm use $TEST_NODE_VERSION;
-    node --version;
-    rm -rf ./node_modules package-lock.json;
-    npm install;
-    make test-without-build;
-    nvm use $TRAVIS_NODE_VERSION;
-    fi
-- if [ "$COVER" = "1" ]; then
-    set -e;
-    make build-node;
-    nvm use $TEST_NODE_VERSION;
-    node --version;
-    rm -rf ./node_modules package-lock.json;
-    npm install;
-    npm run coveralls;
-    nvm use $TRAVIS_NODE_VERSION;
-    fi
+- if [ "$CROSSDOCK" != "1" ]; then make test-without-build; fi
+- if [ "$COVER" = "1" ]; then npm run coveralls; fi
 - if [ "$CROSSDOCK" = "1" ]; then make crossdock-fresh ; fi
+
+after_script:
+- nvm use $TRAVIS_NODE_VERSION
 
 after_failure:
 - if [ "$CROSSDOCK" = "1" ]; then timeout 5 make crossdock-logs ; fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -64,13 +64,15 @@ after_failure:
 - if [ "$CROSSDOCK" = "1" ]; then timeout 5 make crossdock-logs ; fi
 
 after_success:
-- export REPO=jaegertracing/xdock-node
-- export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
-- export TAG=`if [ "$BRANCH" == "master" ]; then echo "latest"; else echo $BRANCH; fi`
-- echo "TRAVIS_BRANCH=$TRAVIS_BRANCH, REPO=$REPO, PR=$PR, BRANCH=$BRANCH, TAG=$TAG"
-- export DOCKER=$(if [ "$BRANCH" == "master" ] && [ "$CROSSDOCK" == "1" ]; then echo docker; else echo true; fi)
-- $DOCKER login -u $DOCKER_USER -p $DOCKER_PASS
-- $DOCKER build -f crossdock/Dockerfile -t $REPO:$COMMIT .
-- $DOCKER tag $REPO:$COMMIT $REPO:$TAG
-- $DOCKER tag $REPO:$COMMIT $REPO:travis-$TRAVIS_BUILD_NUMBER
-- $DOCKER push $REPO
+- if [ "$CROSSDOCK" = "1"] && ["$TRAVIS_PULL_REQUEST" == false]; then
+    export REPO=jaegertracing/xdock-node;
+    export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi);
+    export TAG=`if [ "$BRANCH" == "master" ]; then echo "latest"; else echo $BRANCH; fi`;
+    echo "TRAVIS_BRANCH=$TRAVIS_BRANCH, REPO=$REPO, PR=$PR, BRANCH=$BRANCH, TAG=$TAG";
+    export DOCKER=$(if [ "$BRANCH" == "master" ] && [ "$CROSSDOCK" == "1" ]; then echo docker; else echo true; fi);
+    $DOCKER login -u $DOCKER_USER -p $DOCKER_PASS;
+    $DOCKER build -f crossdock/Dockerfile -t $REPO:$COMMIT .;
+    $DOCKER tag $REPO:$COMMIT $REPO:$TAG;
+    $DOCKER tag $REPO:$COMMIT $REPO:travis-$TRAVIS_BUILD_NUMBER;
+    $DOCKER push $REPO;
+    fi

--- a/Makefile
+++ b/Makefile
@@ -14,12 +14,19 @@ test: build-node
 	npm run test-all
 	npm run check-license
 
+.PHONY: test-without-build
+test-without-build:
+	npm run flow
+	npm run lint
+	npm run test-all
+	npm run check-license
+
 .PHONY: build-node
 build-node: node_modules
 	rm -rf ./dist/
-	node_modules/.bin/babel --presets es2015 --plugins transform-class-properties --source-maps -d dist/src/ src/
-	node_modules/.bin/babel --presets es2015 --plugins transform-class-properties --source-maps -d dist/test/ test/
-	node_modules/.bin/babel --presets es2015 --plugins transform-class-properties --source-maps -d dist/crossdock/ crossdock/
+	node_modules/.bin/babel --presets env --plugins transform-class-properties --source-maps -d dist/src/ src/
+	node_modules/.bin/babel --presets env --plugins transform-class-properties --source-maps -d dist/test/ test/
+	node_modules/.bin/babel --presets env --plugins transform-class-properties --source-maps -d dist/crossdock/ crossdock/
 	cp -R ./test/thrift ./dist/test/thrift/
 	cp package.json ./dist/
 	cp -R ./src/jaeger-idl ./dist/src/

--- a/Makefile
+++ b/Makefile
@@ -9,10 +9,7 @@ publish: build-node
 
 .PHONY: test
 test: build-node
-	npm run flow
-	npm run lint
-	npm run test-all
-	npm run check-license
+	test-without-build
 
 .PHONY: test-without-build
 test-without-build:

--- a/crossdock/docker-compose.yml
+++ b/crossdock/docker-compose.yml
@@ -11,7 +11,7 @@ services:
             - node
         environment:
             - WAIT_FOR=test_driver,go,java,python,node
-            - WAIT_FOR_TIMEOUT=60s
+            - WAIT_FOR_TIMEOUT=240s
 
             - CALL_TIMEOUT=60s
 

--- a/package.json
+++ b/package.json
@@ -25,7 +25,7 @@
     "babel-istanbul": "^0.11.0",
     "babel-plugin-transform-class-properties": "^6.11.5",
     "babel-plugin-transform-flow-strip-types": "^6.8.0",
-    "babel-preset-es2015": "^6.13.2",
+    "babel-preset-env": "^1.6.1",
     "babel-register": "^6.11.6",
     "beautify-benchmark": "^0.2.4",
     "benchmark": "^2.1.1",

